pytest_doctestplus/plugin.py

```diff
@@ -519,159 +519,299 @@ def parse(self, s, name=None):
+                parts = str(file_path).split(os.path.sep)
 
-            __doctest_skip__ = ['ClassName.*']
+                # Don't test files that start with a _
+                if file_path.name.startswith('_'):
+                    return None
 
-        You may also use the string ``'.'`` in ``__doctest_skip__`` to refer
-        to the module itself, in case its module-level docstring contains
-        doctests.
+                # Don't test files in directories that start with a '_' if those
+                # directories are inside docs. Note that we *should* allow for
+                # example /tmp/_q/docs/file.rst but not /tmp/docs/_build/file.rst
+                # If we don't find 'docs' in the path, we should just skip this
+                # check to be safe. We also want to skip any api sub-directory
+                # of docs.
+                if 'docs' in parts:
+                    # We index from the end on the off chance that the temporary
+                    # directory includes 'docs' in the path, e.g.
+                    # /tmp/docs/371j/docs/index.rst You laugh, but who knows! :)
+                    # Also, it turns out lists don't have an rindex method. Huh??!!
+                    docs_index = len(parts) - 1 - parts[::-1].index('docs')
+                    if any(x.startswith('_') or x == 'api' for x in parts[docs_index:]):
+                        return None
+
+                # TODO: Get better names on these items when they are
+                # displayed in py.test output
+                return self._doctest_textfile_item_cls.from_parent(parent, path=file_path)
+
+else:
+
+    class DoctestPlus:
+        def __init__(self, doctest_module_item_cls, doctest_textfile_item_cls, file_globs):
+            """
+            doctest_module_item_cls should be a class inheriting
+            `pytest.doctest.DoctestItem` and `pytest.File`.  This class handles
+            running of a single doctest found in a Python module.  This is passed
+            in as an argument because the actual class to be used may not be
+            available at import time, depending on whether or not the doctest
+            plugin for py.test is available.
+            """
+            self._doctest_module_item_cls = doctest_module_item_cls
+            self._doctest_textfile_item_cls = doctest_textfile_item_cls
+            self._file_globs = file_globs
+            # Directories to ignore when adding doctests
+            self._ignore_paths = []
+
+        def pytest_ignore_collect(self, path, config):
+            """
+            Skip paths that match any of the doctest_norecursedirs patterns or
+            if doctest_only is True then skip all regular test files (eg test_*.py).
+            """
+            if PYTEST_GE_8_0:
+                dirpath = Path(path).parent
+                collect_ignore = config._getconftest_pathlist("collect_ignore",
+                                                              path=dirpath)
```

Comment:
no need for this conditional

Reply:
Good catch! Accidentally left this in from an earlier attempt.